Vcd 11 - Provide a Docker Compose file

monda07 opened the pull request:
Add a docker-compose.yml file to simply all operations for those that are aware of docker-compose
Fix a minor problem with the editing of the config.dat file that will cause imadb queries to fail
Clean up the vector tarball to save space in the image

monda07 committed:
Add Visual Studio Code default registry path

monda07 committed:
Disable the management server in the container. (It is huge)
Remove the initial installer archive to reduce the image size

monda07 committed:
Merge branch 'master' into VCD-X

* master:
  Updated tag in dockerfile comment
  Update README.md

monda07 committed:
Create a docker compose control file for easy launching

monda07 committed:
Merge branch 'VCD-10' into VCD-11

* VCD-10:
  Disable the management server in the container. (It is huge) Remove the initial installer archive to reduce the image size
  Updated tag in dockerfile comment
  Update README.md
  Add Visual Studio Code default registry path

monda07 committed:
Need a "global" replace of the hostname otherwise IMADB queries fail because of a bad "local_vnode" setting
Have the compose file tag the image consistently with the instructions.  Add a note about the Compose file available to the readme

monda07 committed:
Merge branch 'master' into VCD-11

* master:
  Update Dockerfile

# Conflicts:
#	Dockerfile

monda07 committed:
Community edition no longer exists.   Use the standard enterprise version.

clach04 commented:
👍 Looks good, I'd like to get this in.

As agreed in #11, this PR is on hold until #11 is submitted and then @monda07 will resolve conflicts so this can then be merged. I will be happy to approve and merge unless anyone has objections.

clach04 commented:
#11 now merged, so this now conflicts.

monda07 committed:
Merge branch 'master' into VCD-11
Also fixed the Dockerfile as the file pattern did not match downloads from ESD.
Updated the documentation a bit more.  For example Docker no longer has Kitematic.

* master:
  Adapt documentation to include also non-community Vector. (#14)

# Conflicts:
#	Dockerfile

monda07 commented:
Master merged with some additional fixes

clach04 commented:
Making sure I understand this:
* Add Visual Studio Code default registry path
* Disable the management server in the container
* Adds Docker compose file
* Misc doc updates

Do the ports that get exposed include the management server port? Does that need updating? We probably should have a note in the readme about that.

Looking good. Thoughts @fgropengieser?

fgropengieser commented on Dockerfile:
```diff
@@ -44,11 +44,13 @@ RUN cd $VECTOR_ARCHIVE && ./install.sh -express $II_SYSTEM VW -noad && hostname
 
 # fix hostname as localhost
-RUN source /etc/profile.d/vectorVW.sh && sed -i -e "s,`cat /tmp/hostname.build`,localhost,gI" $II_SYSTEM/ingres/files/config.dat && ingsetenv II_HOSTNAME localhost
+RUN source /etc/profile.d/vectorVW.sh && sed -isaved -e "s,`cat /tmp/hostname.build`,localhost,gI" -e 's/mgmtsvr:.*/mgmtsvr: 0/' $II_SYSTEM/ingres/files/config.dat && ingsetenv II_HOSTNAME localhost
```
I do not understand, why this change is required.

monda07 replied:
Which one?  ;-)
-isaved causes the in-place edit by sed to retain a backup file of the config.dat just because that seems like a good idea, especially in diagnostic or debugging.
The management server is a massive component and should not be included in a container.  I was attempting to make the container at least slightly more svelte.....

fgropengieser replied:
I am not familiar with all Ingres components, but does the Vector container work without the mgmtsvr?

clach04 replied:
Yes, Vector will work without mgmtsvr - its only used by the Director (client).

Should someone want to add/restore this they can do so by tweaking the Dockerfile (i.e. not worth making a parameter).

I'm in favor of this

monda07 replied:
A better long term solution would be to make the management server a stand alone container but yes, it is easy for someone to modify the Dockerfile if that was an issue. (There is no scenario I can think of where someone would do this ;-)

fgropengieser commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
From my point of view, VSCode settings should not go into the repository.

monda07 replied:
This is where Visual Studio put the file by default.   I don't use VS so I have no idea what the best practices are.  This was just the result of using VS on this project.

fgropengieser replied:
Yeah, they are put into the hidden folder in you local working copy. However, you should not add it to the commit list. Rather you should add it to gitignore.

clach04 replied:
In that case, I agree remove (and potentially add to `.gitignore`. I was assuming you felt strongly about including it, I wasn't keen but not enough to reject.

fgropengieser commented on docker-compose.yml:
```diff
@@ -0,0 +1,25 @@
+
+services:
+  vector:
+    image: actian/vector:community
```
Does this imply that we have the image pushed to our private Actian docker registry? If so, it should go into documentation. Another question: I thought, we do not offer community edition anymore. So this should then be removed, right?

monda07 replied:
This file is meant to be an example mostly and no, there was no intention to use hub.docker.com at this time.  Since this project builds the image, it will be local to the users environment.  Personally I think it would be cool to have a Vector image publicly available on Docker Hub if we could deal with the licensing issues.

In any case, :community is a bug because the Dockerfile was modified to remove community references and I neglected to remove it here after the merge.

monda07 committed:
Remove Visual Studio code stuff and ignore it for future users.
The :community tag was removed in the Dockerfile but still in the compose file.

clach04 commented:
👍 lgtm :)

monda07 commented:
Updates pushed.

clach04 commented:
squashed, customized commit, merged

monda07 commented:
Thanks everyone for the great feedback!
